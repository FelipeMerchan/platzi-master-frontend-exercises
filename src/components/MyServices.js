import React from 'react'

import './styles/MyServices.css'

function MyServices() {
  return (
    <article className="MyServices">
      <h2 className="MyServices__title">Servicios
        <br /> <span>destacados</span>
      </h2>
      <ul className="MyServices__details">
        <li>
          <p>
            Lorem Ipsum servicio destacado
          </p>
        </li>
        <li>
          <p>
            Otro servicio que debemos destacar
          </p>
        </li>
        <li>
          <p>
            Aqu√≠ va otro servicio
          </p>
        </li>
      </ul>
    </article>
  )
}

export default MyServices