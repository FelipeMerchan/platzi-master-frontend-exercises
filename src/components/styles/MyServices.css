.MyServices {
  font-family: var(--font-4);
  color: #07080d;
}

.MyServices__title {
  margin-inline-start: var(--space-4);
  margin-block-end: var(--space-2);
  font-size: 40px;
  color: #c5345c;
  line-height: .7;
}

.MyServices__title>span {
  color: #66615f;
  padding-inline-start: var(--space-5);
}

.MyServices__details>li {
  margin-block-end: -5px;
  list-style: none;
}

.MyServices__details>li:last-child:before {
  -webkit-box-shadow: 0px 5px 5px 0px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 0px 5px 5px 0px rgba(0, 0, 0, 0.75);
  box-shadow: 0px 5px 5px 0px rgba(0, 0, 0, 0.75);
}

.MyServices__details>li:before {
  content: 'âœ”';
  display: inline-block;
  width: 20px;
  height: 20px;
  margin-inline-end: var(--space);
  border-radius: 50%;
  text-align: center;
  line-height: 20px;
  color: #ffffff;
  background-color: #c5345c;
}

.MyServices__details p {
  display: inline-block;
}