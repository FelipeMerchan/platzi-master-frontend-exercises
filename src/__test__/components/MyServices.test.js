import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import MyServices from '../../components/MyServices'

describe('Probar el componente <MyServices />', () => {
  const myServices = shallow(<MyServices />)
  test('Render del componente MyServices', () => {
    expect(myServices.length).toEqual(1)
  })
  test('Render con el className MyServices__title', () => {
    expect(myServices.exists('.MyServices__title')).toBeTruthy()
  })
  test('Render de los 3 servicios destacados', () => {
    expect(myServices.find('.MyServices__details').children()).toHaveLength(3);
  });
})

describe('MyServices Snapshot', () => {
  const myServices = create(<MyServices />)
  test('Comprobar la UI del componente MyServices', () => {
    expect(myServices.toJSON()).toMatchSnapshot()
  })
})