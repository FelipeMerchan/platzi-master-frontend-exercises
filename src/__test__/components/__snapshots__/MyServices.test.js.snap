// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyServices Snapshot Comprobar la UI del componente MyServices 1`] = `
<article
  className="MyServices"
>
  <h2
    className="MyServices__title"
  >
    Servicios
    <br />
     
    <span>
      destacados
    </span>
  </h2>
  <ul
    className="MyServices__details"
  >
    <li>
      <p>
        Lorem Ipsum servicio destacado
      </p>
    </li>
    <li>
      <p>
        Otro servicio que debemos destacar
      </p>
    </li>
    <li>
      <p>
        Aqu√≠ va otro servicio
      </p>
    </li>
  </ul>
</article>
`;
